File: src/ImeCaretIndicator.Core/IndicatorView.cs
using System.Drawing;

namespace ImeCaretIndicator.Core;

/// <summary>Decide의 출력 — 인디케이터를 그리는 데 필요한 평범한 값.</summary>
public readonly record struct IndicatorView(bool Visible, string Label, Point Position)
{
    public static readonly IndicatorView Hidden = new(false, string.Empty, Point.Empty);
}

File: macos/Sources/CarrotIMECore/IndicatorView.swift
import CoreGraphics

/// decide의 출력 값 객체 — 인디케이터를 어떻게 그릴지. position은 top-left 전역 좌표.
public struct IndicatorView: Equatable {
    public let visible: Bool
    public let label: String
    public let position: CGPoint

    public init(visible: Bool, label: String, position: CGPoint) {
        self.visible = visible
        self.label = label
        self.position = position
    }

    public static let hidden = IndicatorView(visible: false, label: "", position: .zero)
}
